relational_operators/DeleteOperator.cpp

```diff
@@ -81,19 +85,25 @@ void DeleteWorkOrder::execute(QueryContext *query_context,
 
   // TODO(harshad): Stream the block ID only if the predicate returned at least
   // one match in the StorageBlock.
-  ForemanMessage message(ForemanMessage::DataPipelineMessage(delete_operator_index_,
-                                                             input_block_id_,
-                                                             rel_id_));
+  serialization::DataPipelineMessage proto;
+  proto.set_operator_index(delete_operator_index_);
+  proto.set_block_id(input_block_id_);
+  proto.set_relation_id(rel_id_);
+
+  const std::size_t proto_length = proto.ByteSize();
+  char *proto_bytes = static_cast<char*>(std::malloc(proto_length));
```

Comment:
@zuyu I think in general, we shouldn't have a response that the reason is because it was done elsewhere in the code, as going forward we want to identify issues and not only not repeat them, but also fix existing issues. 

BTW, just right after this we do malloc for proto_bytes, so there is an inconsistency in what we allocated on the stack vs the heap.

In general, its a good idea to not use too much stack memory. But, have to be careful to delete allocations on the heap, especially when there are errors/early return.